services/brig/src/Brig/Options.hs

```diff
@@ -313,9 +309,50 @@ data TurnOpts = TurnOpts
     --   should be fetched, in seconds
     configTTL :: !Word32
   }
-  deriving (Show, Generic)
+  deriving (Show)
+
+instance FromJSON TurnOpts where
+  parseJSON = A.withObject "TurnOpts" $ \o ->
+    TurnOpts
+      <$> A.parseJSON (A.Object o)
+      <*> o .: "secret"
+      <*> o .: "tokenTTL"
+      <*> o .: "configTTL"
+
+data TurnServersSource
+  = TurnSourceDNS TurnDnsOpts
+  | TurnSourceFiles TurnServersFiles
+  deriving (Show)
+
+instance FromJSON TurnServersSource where
+  parseJSON v =
+    -- TODO: print better errors on failure
+    (TurnSourceDNS <$> A.parseJSON v)
+      <|> (TurnSourceFiles <$> A.parseJSON v)
```

Comment:
Hm, this is very weird. This implicitly tries to parse two options, and if they are not provided, parses two other options. That's very hard for anyone to figure out who cannot read Haskell source code. Can we have a proper enum config option that can be explicitly set? `turnServersSource: "files" # "files" | "dns"` ? We can default this to "files" inside helm charts brig/values.yaml for backwards compatibility.